### PR TITLE
use c literals in compiler and library

### PR DESCRIPTION
**Pull request by klensy**
Use c literals #108801 in compiler and library

currently blocked on:
* <strike>rustfmt: don't know how to format c literals</strike> nope, nightly one works.
* <strike>bootstrap</strike>

r? @ghost
@rustbot blocked

### CODE REVIEW

File: compiler/rustc_codegen_llvm/src/builder.rs [hunk 1]
```diff
@@ -45,8 +44,7 @@ impl Drop for Builder<'_, '_, '_> {
     }
 }
 
-// FIXME(eddyb) use a checked constructor when they become `const fn`.
-const EMPTY_C_STR: &CStr = unsafe { CStr::from_bytes_with_nul_unchecked(b"\0") };
+const EMPTY_C_STR: &CStr = c"";
```
Comment (fmease) [inline compiler/rustc_codegen_llvm/src/builder.rs:47]: this could be inlined into its only use-site in `UNNAMED`

**Comment by klensy** [comment]
r? compiler

**Pull request merged**